### DISCUSSION TITLE
웹 송신기 -> 앱 수신기 에서 수신 속도 관련 문제

### DISCUSSION THREAD

**Discussion by sunyeongan** [euphony-io/euphony#175] [Q&A]
안녕하세요! 
저희팀은 웹에서 음파신호를 발생시키고 앱에서 음파신호를 받고 있습니다. 
웹에서 보내는 문자열이 길어지면, 앱에서 문자열을 수신하는 과정에서 
점점 수신 정확도가 떨어지거나 속도가 저하되는 상황이 발생하였습니다.
https://github.com/orgs/euphony-io/discussions/155
위의 질문에서 답변은 
웹에서 문자열을 전송할 때 문자열은 무한대의 길이로 보낼 수 있지만 음파신호와 노이즈가 섞여서 올바른 데이터로 처리할 수  없다는 것을 알게되었습니다. 

그렇다면 안드로이드 수신기에서 문자열을 받는 속도개선, 수신 정확도를 개선하기 위해서 
https://github.com/orgs/euphony-io/discussions/52
위의 디스커션을 읽어보았습니다. 

노이즈를 제거 하기 위해서 어떤 방법을 더 생각해봐야할지 조언주시면 감사하겠습니다..!

**Comment by designe** [comment]
안녕하세요. 선영님 :)
선영님은 통신 이론을 알고 계셔서 아무래도 다양한 노이즈 제거 방법을 생각해보셨을 것 같은데요!

사실 euphony는 현재 무한대의 길이의 컨텐츠를 하나의 패킷으로만 전송하는 것이 가장 큰 문제점입니다.
당연히 데이터의 길이가 길어지면 오류가 많이 섞여 있을 것이고 이를 보정하기 위한 첫번째 시작점은 멀티 패킷 구현입니다.

현재는 싱글 패킷으로만 구현되어 있는데 이를 잘개 여러 멀티 패킷으로 나누어 전송하면 길더라도 성공률이 높을 수 있습니다.

다음으로 필요한 것은 바로 리드 솔로몬 같은 https://en.wikipedia.org/wiki/Reed%E2%80%93Solomon_error_correction
오류 복원 알고리즘입니다. 멀티 패킷에 숨어있는 오류들을 다른 패킷으로 복원하는 알고리즘을 적용하면 또 한번 성능은 좋아집니다.

이외 링크주신 discussions 을 포함하여 여러 방법이 있어요!
